sdks/python/apache_beam/io/filebasedsink.py

```diff
@@ -309,11 +341,18 @@ def _template_to_format(shard_name_template):
                        shard_name_template)
     shard_name_format = shard_name_template.replace(
         m.group(0), '%%(shard_num)0%dd' % len(m.group(0)))
-    m = re.search('N+', shard_name_format)
-    if m:
-      shard_name_format = shard_name_format.replace(
-          m.group(0), '%%(num_shards)0%dd' % len(m.group(0)))
-    return shard_name_format
+    return FileBasedSink._template_replace_num_shards(shard_name_format)
+
+  @staticmethod
+  def _template_to_glob_format(shard_name_template):
+    if not shard_name_template:
+      return ''
+    m = re.search('S+', shard_name_template)
+    if m is None:
+      raise ValueError("Shard number pattern S+ not found in template '%s'" %
```

Comment:
Not sure if this error message is useful for an end-user since it might not be immediately clear what S+ is. Can we elaborate more ?

Reply:
There's documentation for `shard_name_template` in the constructors for classes such as `WriteToText`.
I added an explicit reference to `shard_name_template`.
I could remove the `+` if you think it'd make it clearer.